### DISCUSSION TITLE
Noob's question: Change wakeword of a headless sepia client

### DISCUSSION THREAD

**Discussion by ombonbon** [SEPIA-Framework/sepia-docs#217] [Q&A] [answered]
Hallo @fquirin , first of all thank you so much for your great SEPIA project. I have installed one server in a proxmox container running debian 11 and now trying to explore more by installing headless clients on 2 raspberry pi.
Now comes the question :)
I am trying to change the wakeword without any success. I change it in the settings.js file in ~/clexi/www/sepia. Is the file location correct?
My setting looks like 
"wakeWordNames": "Jarvis",
"wakeWordVersion": "1.9",
"voiceEngine": "sepia",
"voiceCustomServerURI": "",

Sorry for the very basic question, by the way....

**Comment by fquirin** [comment, marked as answer]
Hi @ombonbon 🙂 ,

that looks correct so far, but you might need to check the ["useWakeWord"](https://github.com/SEPIA-Framework/sepia-html-client-app/blob/master/Settings.md#wake-word-settings) field a bit further down.
Btw the field `"voiceEngine": "sepia"` refers only to the speech-to-text engine (=SEPIA STT-Server), not the wake-word, just in case you were wondering. 

To debug the wake-word it can be useful to connect to the DIY client via the remote-terminal (SEPIA Control-HUB) first and look for error messages.
If you're running a desktop client as well you can set it up properly and then check settings page 2 for "export" to see the equivalent JSON configuration.

**Reply by ombonbon** [reply]
I set "useWakeWord" to true.
As you suggested, I changed the settings.js file and set "wakeWordNames" to "Jarvis" also set "wakeWordVersion" to "1.9". Then I 'call reload' from SEPIA Control-HUB. Here is the error message I got:
Broadcaster event: {"broadcast":{"client":"o2_chrome_app_v0.25.0","deviceId":"o2","sepia-wake-word":{"state":"error","msg":"Uncaught RangeError: WebAssembly.Memory(): could not allocate memory"}}}
I will try your next suggestion to change the client mode to "display" (via menu number 4 when running setup.sh)

**Reply by fquirin** [reply]
Oh, is that a Raspberry Pi 3 or a Pi 4 with 1GB RAM? Unfortunately there is a bug in the 1.9er wake-word engine that leads to unnecessarily high memory consumption 😬. You can try the 2.1 version (if you are ok with creating a free Porcupine account to get a key) or an older version with a different wake-word.

**Reply by ombonbon** [reply]
Ah so....sorry for not mentioning that in the beginning. Yes, it is a Raspberry Pi 3.
After creating a free Porcupine account and using version 2.1, the wake-word is working well. Thank you so much @fquirin for your support and everything you have done with this SEPIA framework. You're rock man!
Oh, for anyone having the same problem, just do as @fquirin said, create a free Porcupine account [here](https://console.picovoice.ai/signup), and after confirming your email and completing the account information (it's just asking to provide our first and last name), you will see your "Access Key" which you can copy and paste to the SEPIA Control-HUB.
